src/pykeen/triples/triples_factory.py

```diff
@@ -439,7 +476,7 @@ def with_labels(
             if not existing_ids.issubset(new_labeling.values()):
                 diff = existing_ids.difference(new_labeling.values())
                 raise ValueError(f"Some existing IDs do not occur in the new {name} labeling: {diff}")
-        return TriplesFactory(
+        return self.new(
```

Comment:
in this case, you can also do `self.__class__()`

Reply:
Oh, yes 🙈

Reply:
https://github.com/pykeen/pykeen/pull/1416/commits/004178f7d7d5da1c7b6b0ec59d250803083772e4